### PR TITLE
Belindas-closet-5-3-update-ci-pipeline

### PR DESCRIPTION
**Pull request by taylorpapke**
resolves: https://github.com/SeattleColleges/belindas-closet-nextjs/issues/3

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,17 +15,12 @@ jobs:
...
 
-    strategy:
-      matrix:
-        node-version: [16.x, 18.x] # 19.x stalling on seattlecolleges repo but running fine on my forked repo..
-        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
-
     steps:
       - uses: actions/checkout@v3
-      - name: Use Node.js ${{ matrix.node-version }}
+      - name: Use Node.js 16
```
Comment (bcko) [inline .github/workflows/ci.yml:20]: Current LTS version is 18, so let's bump the nodejs version from 16 to 18. 
Screenshot 2023-08-31 at 2 44 23 PM

**Review by bcko — DISMISSED** [review]

**Comment by bcko** [comment]
@taylorpapke 
I see that you added html, css, javascript files under `coverage`. 
can you specify where you are getting the html, css, javascript files from so we can come up with a plan to maintain the code?

**Review by bcko — DISMISSED** [review]

**Comment by bcko** [comment]
@theGaryLarson
can you specify where you are getting the html, css, javascript files from so we can come up with a plan to maintain the code?

File: .github/workflows/on-pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+name: CI (on pull-request)
+run-name: PR from branch ${{ github.ref }}
+
+# Will run CI workflow on an opened or reopened PR for easy reference. For compilation, unit tests, code coverage etc.
+on:
+  pull_request:
+    types: [opened, reopened]
+jobs:
+  Continuous-Integration:
+    uses: ./.github/workflows/ci.yml
```
Comment (theGaryLarson) [inline .github/workflows/on-pull-request.yml:15]: Add name: on pull request. This is required to set it as a required status check
see: https://github.com/SeattleColleges/nsc-events-nextjs/blob/main/.github/workflows/on-pull-request.yml

File: .github/workflows/on-push-staging.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  pull_request:
+    branches: [ "staging" ]
+
+    types:
+      - closed
+
+jobs:
+  # redundant check to ensure CI passes before deployment
+  CI:
+    if: github.event.pull_request.merged == true || github.event_name == 'push'
```
Comment (theGaryLarson) [inline .github/workflows/on-push-staging.yml:34]: add name. See nsc-events (next.js) files. https://github.com/SeattleColleges/nsc-events-nextjs/blob/main/.github/workflows/on-push-staging.yml

File: .github/workflows/on-new-branch.yml [hunk 4]
```diff
@@ -16,18 +12,15 @@ run-name: CI On New Branch
...
   create:
 
-  pull_request:
-    branches: [ "main" ]
-
-    types:
-      - opened
-
 jobs:
   CI:
```
Comment (theGaryLarson) [inline .github/workflows/on-new-branch.yml:25]: add name see nsc-events next file 
https://github.com/SeattleColleges/nsc-events-nextjs/blob/main/.github/workflows/on-new-branch.yml

File: .github/workflows/on-push-staging.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    uses: ./.github/workflows/ci.yml # reusable workflow file path
+
+
+  CD:
+    if: github.event.pull_request.merged == true || github.event_name == 'push'
```
Comment (theGaryLarson) [inline .github/workflows/on-push-staging.yml:39]: same as above.

File: .github/workflows/sync-staging.yml [hunk 6]
```diff
@@ -56,4 +50,4 @@ jobs:
           git pull
           git checkout staging
           git merge main --no-ff
-          git push origin staging
+          git push origin staging
```
Comment (theGaryLarson) [inline .github/workflows/sync-staging.yml:53]: this is minor but a more descriptive name here. 
https://github.com/SeattleColleges/nsc-events-nextjs/blob/main/.github/workflows/sync-staging.yml

**Review by theGaryLarson — CHANGES_REQUESTED** [review]
Looks good overall. Some minor changes included in code comments below.
The job names are an important add. Having the names is required to set required status checks for GitHub actions.

**Comment by theGaryLarson** [comment]
@bcko, the html and related files are being generated by jest --coverage in the ci.yml. I have not had the chance to look into what types of options there are, if any, to maintain those files.

File: .github/workflows/sync-staging.yml (hunk 6 above)
Comment (taylorpapke) [reply .github/workflows/sync-staging.yml:53]: Is this for the name of the job? The link has the same code. Is there any more context for this?

Comment (brinkbrink) [reply .github/workflows/sync-staging.yml:53]: I think he's referencing the run-name?

What you have here:
`run-name: Sync-Staging`
What Gary has there:
`run-name: Merge main into Staging`

**Review by theGaryLarson — APPROVED** [review]
Looks good, thanks for adding those names, appreciate it!

**Review by bcko — APPROVED** [review]

**Pull request merged**